pkg/resolution/resource/crd_resource_test.go

```diff
@@ -0,0 +1,260 @@
+		},
+		{
+			name:                     "resolution request exist and status is false",
+			inputRequest:             getResolutionRequestFromFile("testdata/submit.resolution_request.false.yaml"),
+			expectedRequest:          nil,
+			expectedResolvedResource: nil,
+			expectedErr:              resolutioncommon.NewError(resolutioncommon.ReasonResolutionFailed, errors.New("message")),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			d := test.Data{}
+			if tc.inputRequest != nil {
+				d.ResolutionRequests = []*v1beta1.ResolutionRequest{tc.inputRequest}
+			}
+
+			testAssets, cancel := getCRDRequester(t, d)
+			defer cancel()
+			ctx := testAssets.Ctx
+			clients := testAssets.Clients
+
+			resolver := ResolverName("git")
+			request := getRequestFromFile("testdata/request.golden.yaml")
+			crdRequester := NewCRDRequester(clients.ResolutionRequests, testAssets.Informers.ResolutionRequest.Lister())
+			resolvedResource, err := crdRequester.Submit(ctx, resolver, request.Request())
+
+			// check the error
+			if (tc.expectedErr != nil && err == nil) ||
+				(tc.expectedErr == nil && err != nil) {
+				t.Fatalf("expected error %v, but got %v", tc.expectedErr, err)
+			} else if tc.expectedErr != nil {
+				if d := cmp.Diff(tc.expectedErr.Error(), err.Error()); d != "" {
+					t.Fatalf("expected error to match %s", diff.PrintWantGot(d))
+				}
+			}
+
+			// check the resolved resource
+			if (tc.expectedResolvedResource != nil && resolvedResource == nil) ||
+				(tc.expectedResolvedResource == nil && resolvedResource != nil) {
+				t.Fatalf("expected resolved resource equal %v, but got %v", tc.expectedResolvedResource, resolvedResource)
+			} else if tc.expectedResolvedResource != nil {
+				if d := cmp.Diff(tc.expectedResolvedResource.req, resolvedResource.(readOnlyResolutionRequest).req); d != "" {
+					t.Fatalf("expected resolved resource to match %s", diff.PrintWantGot(d))
+				}
+			}
+
+			// check the resolution request
+			if tc.expectedRequest != nil {
+				resolutionrequest, err := clients.ResolutionRequests.ResolutionV1beta1().
+					ResolutionRequests(request.Namespace).Get(ctx, request.Name, metav1.GetOptions{})
+				if err != nil {
+					t.Fatalf("unexpected error getting resource requests: %v", err)
+				}
+				if d := cmp.Diff(tc.expectedRequest, resolutionrequest); d != "" {
+					t.Fatalf("expected resolution request to match %s", diff.PrintWantGot(d))
+				}
+			}
+		})
+	}
+}
+
+// getResolutionRequestFromFile returns a resolution request that loads data from a file.
+func getResolutionRequestFromFile(filepath string) *v1beta1.ResolutionRequest {
+	return test.LoadFromFileToObject(filepath, &v1beta1.ResolutionRequest{}).(*v1beta1.ResolutionRequest)
+}
+
+// getRequestFromFile returns a raw request that loads data from a file.
+func getRequestFromFile(filepath string) *test.RawRequest {
+	return test.LoadFromFileToObject(filepath, &test.RawRequest{}).(*test.RawRequest)
```

Comment:
I'll leave this up to you, but I prefer to have the input test data defined in the test case instead of read from a file. I think the test cases are easier to understand when you only have to look in one spot at a time.

Reply:
I still prefer to have the test data defined alongside the test cases, but won't block on this

Reply:
Sure, I'll try to follow your existing agreements as much as possible.

Normally, I can quickly jump to a specified file in an editor, but it's inconvenient when doing code review on GitHub.

Reply:
I refactored this part by following the `mustParse` style of the community, and now we don’t need to load test cases from files anymore.